### PR TITLE
Task-49284: copy/past problem in chat text area

### PR DESCRIPTION
**Pull request by wassimmhamdi**
Problem: The background color is copied also and since the text color became black the message is unreadable
Fix: The background color shouldn’t be copied

### CODE REVIEW

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 1]
```diff
@@ -220,6 +222,34 @@ export default {
...
+        const textNode = document.createTextNode(text);
+        range.insertNode(textNode);
+        range.selectNodeContents(textNode);
+        range.collapse(false);
+
+        const selection = window.getSelection();
+        selection.removeAllRanges();
+        selection.addRange(range);
+      }
+    },
```
Comment (AzmiTouil) [inline application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: What happens if I copy/paste an image ? it works?

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 2]
```diff
@@ -347,7 +348,29 @@ export default {
...
+      }
+      else {
+        // cancel paste
+        e.preventDefault();
+        // get text representation of clipboard
+        this.text = (e.originalEvent || e).clipboardData.getData('text/plain');
+        // insert text manually
+        document.execCommand('insertHTML', false, this.text);
+      }
+    },
```
Comment (AzmiTouil) [inline application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: Format the code please!

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 3]
```diff
@@ -347,7 +348,29 @@ export default {
...
       return message;
-    }
+    },
+    paste(e) {
+      // consider the first item (can be easily extended for multiple items)
+      const item = e.clipboardData.items[0];
+      //test if the type of item e
+      if (item.type.indexOf('image') === 0)
+      {
+        const blob = item.getAsFile();
```
Comment (AzmiTouil) [inline application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]:
```suggestion
        const pastedImage = item.getAsFile();
```

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 4]
```diff
@@ -347,7 +348,29 @@ export default {
...
+        const reader = new FileReader();
+        reader.onload = function(event) {
+          document.getElementById('container').src = event.target.result;
+        };
+        reader.readAsDataURL(blob);
```
Comment (AzmiTouil) [inline application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]:
```suggestion
        reader.readAsDataURL(pastedImage);
```

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue (hunk 1 above)
Comment (wassimmhamdi) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: yes it works correctly !

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue (hunk 2 above)
Comment (wassimmhamdi) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: code formatted

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue (hunk 3 above)
Comment (wassimmhamdi) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: variable renamed

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue (hunk 4 above)
Comment (wassimmhamdi) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: variable renamed

**Review by AzmiTouil — DISMISSED** [review]

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 5]
```diff
@@ -347,7 +348,27 @@ export default {
...
+    },
+    paste(e) {
+      // consider the first item (can be easily extended for multiple items)
+      const item = e.clipboardData.items[0];
+      //test if the type of item e
+      if (item.type.indexOf('image') === 0) {
+        const pastedImage = item.getAsFile();
+        const reader = new FileReader();
+        reader.onload = function (event) {
+          document.getElementById('container').src = event.target.result;
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: What is this element **container** created ? is it a hidden element ? I think you need to create one that put the image inside ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue (hunk 5 above)
Comment (wassimmhamdi) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: element **container** changed to **messageComposerArea**

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 6]
```diff
@@ -347,7 +348,27 @@ export default {
...
+    },
+    paste(e) {
+      // consider the first item (can be easily extended for multiple items)
+      const item = e.clipboardData.items[0];
+      //test if the type of item e
+      if (item.type.indexOf('image') === 0) {
+        const pastedImage = item.getAsFile();
+        const reader = new FileReader();
+        reader.onload = function (event) {
+          document.getElementById('messageComposerArea').src = event.target.result;
```
Comment (rdenarie) [inline application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: I think you should use this.$refs.messageComposerArea
as it is done elsewhere in this component.

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue (hunk 6 above)
Comment (wassimmhamdi) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: changes done !

**Review by ahamdi — DISMISSED** [review]

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 7]
```diff
@@ -347,7 +348,27 @@ export default {
...
+          this.$refs.messageComposerArea.src = event.target.result;
+        };
+        reader.readAsDataURL(pastedImage);
+      } else {
+        // cancel paste
+        e.preventDefault();
+        // get text representation of clipboard
+        this.text = (e.originalEvent || e).clipboardData.getData('text/plain');
+        // insert text manually
+        document.execCommand('insertHTML', false, this.text);
```
Comment (rdenarie) [inline application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: This command seems deprecated : https://developer.mozilla.org/fr/docs/Web/API/Document/execCommand
We should use something else

**Review by rdenarie — CHANGES_REQUESTED** [review]
****

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue (hunk 7 above)
Comment (wassimmhamdi) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: I change it to other method that do the same work

**Review by rdenarie — APPROVED** [review]

**Pull request merged**